### PR TITLE
Feat/2463 create git pull service with amplcation

### PR DESCRIPTION
**Pull request by abrl91**
#2463 

This task's Purpose was to generate an app with Amplicaton and determined what we need.
Only one table GitRepositoryPull with some properties.

Then I deleted the code that I don't need for my task: graphql, swagger, rest

Removed unused packages from package.json
Changed Dockerfile and docker-compole files

## PR Checklist
- [ ] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -4,7 +4,6 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "amplication",
```
Comment (mshidlov) [inline package-lock.json:7]: this is not suppose to be here

File: packages/amplication-git-pull-remote/package.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "@amplication/git-pull-remote",
```
Comment (mshidlov) [inline packages/amplication-git-pull-remote/package.json:2]: please rename it to git pull service, I do want to have a services naming convention for service names suffix of "-service"

File: packages/amplication-git-pull-service/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+WORKDIR /app
+COPY lerna.json /app
+COPY package*.json /app
+RUN npm ci --production
+
+FROM base AS build
+WORKDIR /app
+# Copy amplication/server and the its dependent packages
+COPY packages/amplication-server packages/amplication-git-pull-remote
```
Comment (abrl91) [inline packages/amplication-git-pull-service/Dockerfile:20]:
```suggestion
COPY packages/amplication-git-pull-service packages/mplication-git-pull-service
```

File: packages/amplication-git-pull-service/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+COPY packages/amplication-container-builder packages/amplication-container-builder
+COPY packages/amplication-deployer packages/amplication-deployer
```
Comment (abrl91) [inline packages/amplication-git-pull-service/Dockerfile:22]:
```suggestion

```

File: packages/amplication-git-pull-service/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+RUN npm run build -- --scope @amplication/amplication-git-pull-service --include-dependencies
+
+# Removes packages/*/node_modules
+# https://github.com/lerna/lerna/issues/2196#issuecomment-994882795
+RUN npm run clean -- --yes
+# Rebuild production node_modules
+RUN npm run bootstrap -- -- --production --scope @amplication/amplication-git-pull-service --include-dependencies
+
+FROM base as server
+WORKDIR /app/packages/amplication-git-pull-remote
```
Comment (abrl91) [inline packages/amplication-git-pull-service/Dockerfile:37]:
```suggestion
WORKDIR /app/packages/amplication-git-pull-service
```

File: packages/amplication-git-pull-service/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+# Copy all distributions and node_modules for amplication/server and its dependencies
+COPY --from=build /app/packages/amplication-server/package.json /app/packages/amplication-git-pull-remote/package.json
+COPY --from=build /app/packages/amplication-server/node_modules /app/packages/amplication-git-pull-remote/node_modules
+COPY --from=build /app/packages/amplication-server/dist /app/packages/amplication-git-pull-remote/dist
```
Comment (abrl91) [inline packages/amplication-git-pull-service/Dockerfile:41]:
```suggestion
COPY --from=build /app/packages/amplication-git-pull-service/package.json /app/packages/amplication-git-pull-service/package.json
COPY --from=build /app/packages/amplication-git-pull-service/node_modules /app/packages/amplication-git-pull-service/node_modules
COPY --from=build /app/packages/aamplication-git-pull-servicer/dist /app/packages/amplication-git-pull-service/dist
```

File: packages/amplication-git-pull-service/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+COPY --from=build /app/packages/amplication-container-builder/package.json /app/packages/amplication-container-builder/package.json
+COPY --from=build /app/packages/amplication-container-builder/node_modules /app/packages/amplication-container-builder/node_modules
+COPY --from=build /app/packages/amplication-container-builder/dist /app/packages/amplication-container-builder/dist
+
+COPY --from=build /app/packages/amplication-deployer/package.json /app/packages/amplication-deployer/package.json
+COPY --from=build /app/packages/amplication-deployer/node_modules /app/packages/amplication-deployer/node_modules
+COPY --from=build /app/packages/amplication-deployer/dist /app/packages/amplication-deployer/dist
```
Comment (abrl91) [inline packages/amplication-git-pull-service/Dockerfile:49]:
```suggestion

```

File: packages/amplication-git-pull-service/README.md [hunk 8]
```diff
@@ -0,0 +1 @@
+# Amplication git pull remote service
```
Comment (abrl91) [inline packages/amplication-git-pull-service/README.md:1]: Amplication git pull service

File: packages/amplication-git-pull-service/docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+      POSTGRES_PASSWORD: ${POSTGRESQL_PASSWORD}
+    volumes:
+      - postgres:/var/lib/postgresql/data
+  scheduler:
+    image: amplication/scheduler
+    command: --request.url http://host.docker.internal:3000/system/update-statuses --request.method POST --schedule '* * * * *'
+    network_mode: host
+volumes:
+  postgres: ~
+
```
Comment (abrl91) [inline packages/amplication-git-pull-service/docker-compose.yml:19]:
```suggestion
version: "3"
services:
  server:
    build:
      context: .
      args:
        NPM_LOG_LEVEL: notice
    ports:
      - "${SERVER_PORT}:3000"
    environment:
      POSTGRESQL_URL: "postgres://${POSTGRESQL_USER}:${POSTGRESQL_PASSWORD}@db:5432"
      BCRYPT_SALT: "${BCRYPT_SALT}"
      JWT_SECRET_KEY: "${JWT_SECRET_KEY}"
      JWT_EXPIRATION: "${JWT_EXPIRATION}"
    depends_on:
      - migrate
  migrate:
    build:
      context: .
      args:
        NPM_LOG_LEVEL: notice
    command: npm run db:init
    working_dir: /app/server
    environment:
      POSTGRESQL_URL: "postgres://${POSTGRESQL_USER}:${POSTGRESQL_PASSWORD}@db:5432"
      BCRYPT_SALT: "${BCRYPT_SALT}"
    depends_on:
      - db
  db:
    image: postgres:12
    environment:
      POSTGRES_USER: ${POSTGRESQL_USER}
      POSTGRES_PASSWORD: ${POSTGRESQL_PASSWORD}
    volumes:
      - postgres:/var/lib/postgresql/data
volumes:
  postgres: ~


```

File: packages/amplication-git-pull-service/package.json [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    "@prisma/client": "3.6.0",
+    "apollo-server-express": "3.5.0",
+    "bcrypt": "5.0.1",
+    "class-transformer": "0.5.1",
+    "class-validator": "0.13.2",
+    "nest-access-control": "2.0.3",
+    "nest-morgan": "1.0.1",
+    "nestjs-prisma": "0.14.3",
+    "npm-run-all": "4.1.5",
+    "octokit": "^1.7.1",
```
Comment (abrl91) [inline packages/amplication-git-pull-service/package.json:38]: npm i --save-exact

File: packages/amplication-git-pull-service/package.json [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+    "prettier": "2.6.0",
+    "reflect-metadata": "0.1.13",
+    "simple-git": "^3.4.0",
```
Comment (abrl91) [inline packages/amplication-git-pull-service/package.json:41]: npm i --save-exact

File: packages/amplication-git-pull-service/package.json [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+    "docker:build": "docker build ."
+  },
+  "dependencies": {
+    "@nestjs/common": "8.2.3",
+    "@nestjs/config": "1.1.5",
+    "@nestjs/core": "8.2.3",
+    "@nestjs/platform-express": "8.2.3",
+    "@nestjs/serve-static": "2.2.2",
+    "@prisma/client": "3.6.0",
+    "apollo-server-express": "3.5.0",
```
Comment (abrl91) [inline packages/amplication-git-pull-service/package.json:30]:
```suggestion
```

File: packages/amplication-git-pull-service/package.json [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+    "bcrypt": "5.0.1",
+    "class-transformer": "0.5.1",
+    "class-validator": "0.13.2",
```
Comment (abrl91) [inline packages/amplication-git-pull-service/package.json:33]: check if needed

File: packages/amplication-git-pull-service/package.json [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+    "nest-access-control": "2.0.3",
+    "nest-morgan": "1.0.1",
+    "nestjs-prisma": "0.14.3",
+    "npm-run-all": "4.1.5",
```
Comment (abrl91) [inline packages/amplication-git-pull-service/package.json:37]: check if needed

File: packages/amplication-git-pull-service/prisma/migrations/20220319155701_initial_version/migration.sql [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+    "owner" TEXT,
+    "branch" TEXT,
+    "status" "EnumGitHubRepositoryStatus",
+    "name" TEXT,
+
+    CONSTRAINT "GitHubRepository_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateIndex
+CREATE UNIQUE INDEX "User_username_key" ON "User"("username");
```
Comment (abrl91) [inline packages/amplication-git-pull-service/prisma/migrations/20220319155701_initial_version/migration.sql:34]: update the schema

File: packages/amplication-git-pull-service/prisma/schema.prisma [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+  pushedAt  DateTime?                   @updatedAt
+  provider  String?
+  owner     String?
+  branch    String?
+  status    EnumGitRepositoryPullStatus?
+  name      String?
+}
+
+enum EnumGitRepositoryPullStatus {
+  Created
```
Comment (abrl91) [inline packages/amplication-git-pull-service/prisma/schema.prisma:34]: update the schema

File: packages/amplication-git-pull-service/src/app.module.ts [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+import { ConfigModule } from "@nestjs/config";
+import { ServeStaticModule } from "@nestjs/serve-static";
+import { ServeStaticOptionsService } from "./serveStaticOptions.service";
+
+@Module({
+  controllers: [],
+  imports: [
+    GitRepositoryPullModule,
+    HealthModule,
+    SecretsManagerModule,
```
Comment (abrl91) [inline packages/amplication-git-pull-service/src/app.module.ts:15]:
```suggestion
```

File: packages/amplication-git-pull-service/src/grants.json [hunk 18]
```diff
@@ -0,0 +1,62 @@
+[
```
Comment (abrl91) [inline packages/amplication-git-pull-service/src/grants.json:1]: delete

File: packages/amplication-git-pull-service/src/serveStaticOptions.service.ts [hunk 19]
```diff
@@ -0,0 +1,39 @@
+import * as path from "path";
```
Comment (abrl91) [inline packages/amplication-git-pull-service/src/serveStaticOptions.service.ts:1]: delete

File: packages/amplication-git-pull-service/src/app.module.ts (hunk 17 above)
Comment (abrl91) [reply packages/amplication-git-pull-service/src/app.module.ts:15]: we decided to leave it for now

File: packages/amplication-git-pull-service/prisma/migrations/20220319155701_initial_version/migration.sql [hunk 20]
```diff
@@ -0,0 +1,34 @@
+-- CreateEnum
```
Comment (mshidlov) [inline packages/amplication-git-pull-service/prisma/migrations/20220319155701_initial_version/migration.sql:1]: DELETE MIGRATIONS

File: packages/amplication-git-pull-service/prisma/schema.prisma [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+  repositoryOwner       String
+  repositoryName        String
+  branch                String
+  commit                String
+  status                EnumGitPullEventStatus
+  pushedAt              DateTime
+  updatedAt             DateTime                    @updatedAt
+  createdAt             DateTime                    @default(now())
+
+  @@index([pushedAt], map: "idx_pushedAt")
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-git-pull-service/prisma/schema.prisma:22]: represents the index names
https://www.prisma.io/docs/concepts/components/prisma-schema/names-in-underlying-database#using-custom-constraint--index-names

File: packages/amplication-git-pull-service/prisma/migrations/20220319155701_initial_version/migration.sql (hunk 20 above)
Comment (abrl91) [reply packages/amplication-git-pull-service/prisma/migrations/20220319155701_initial_version/migration.sql]: done

**Pull request merged**